modules/artifact/src/main/java/org/mule/module/artifact/classloader/IBMMQResourceReleaser.java

```diff
@@ -382,4 +391,51 @@ private void processThreadLocalMap(Field threadLocalMapTableField, Object thread
       }
     }
   }
+
+  /**
+   * Close JMSCCThreadPoolMaster threads pending to close to avoid thread leaks after redeployment
+   */
+  private void disposeMQThreads() {
+    List<Thread> threads = ThreadUtils.getAllThreads().stream()
+        .filter(thread -> thread.getName().equals(JMSCC_THREAD_POOL_MAIN_NAME))
+        .filter(thread -> thread.getClass().getClassLoader() == driverClassLoader)
```

Comment:
Do only one filter. Move both predicates into a function. For example `isThreadToBeDisposed`.